scripts/generate_syncset.py

```diff
@@ -29,31 +32,60 @@ def get_all_yaml_obj(file_paths):
             yaml_objs.append(obj)
     return yaml_objs
 
+def process_yamls(name, directory, obj):
+    o = copy.deepcopy(obj)
+    # Get all yaml files as array of yaml objecys
+    yamls = get_all_yaml_obj(get_all_yaml_files(directory))
+
+    for y in yamls:
+        if 'patch' in y:
+            if not 'patches' in o['spec']:
+                o['spec']['patches'] = []
+            o['spec']['patches'].append(y)
+        else:
+            if not 'resources' in o['spec']:
+                o['spec']['resources'] = []
+            o['spec']['resources'].append(y)
+
+    o['metadata']['name'] = name
+
+    # append object to the template's objects
+    template_data['objects'].append(o)
+
 
 if __name__ == '__main__':
     #Argument parser
     parser = argparse.ArgumentParser(description="selectorsyncset generation tool", usage='%(prog)s [options]')
-    parser.add_argument("--template-path", "-t", required=True, help="Path to template file [required]")
+    parser.add_argument("--template-dir", "-t", required=True, help="Path to template directory [required]")
     parser.add_argument("--yaml-directory", "-y", required=True, help="Path to folder containing yaml files [required]")
     parser.add_argument("--destination", "-d", required=True, help="Destination for selectorsynceset file [required]")
-    parser.add_argument("--git-hash", "-c", required=True, help="Commit hash of commit used [required]")
+    parser.add_argument("--repo-name", "-r", required=True, help="Name of the repository [required]")
     arguments = parser.parse_args()
 
-    # Get all yaml files as array of yaml objecys
-    yamls = get_all_yaml_obj(get_all_yaml_files(arguments.yaml_directory))
     # Get the template data
-    template_data = get_yaml(arguments.template_path)
-    # Configure template
-    for obj in template_data['objects']:
-        if not 'labels' in obj['metadata']:
-            obj['metadata']['labels'] = {}
-        # create labels
-        obj['metadata']['labels']['managed.openshift.io/osd'] = "true"
-        obj['metadata']['labels']['managed.openshift.io/gitHash'] = arguments.git_hash
-
-        # create resources
-        obj['spec']['resources'] = yamls
-    
-    # write selectorsyncset file
+    template_data = get_yaml(os.path.join(arguments.template_dir, "template.yaml"))
+    selectorsyncset_data = get_yaml(os.path.join(arguments.template_dir, "selectorsyncset.yaml"))
+
+    # Create required labels on SelectorSyncSet
+    if not 'labels' in selectorsyncset_data['metadata']:
+        selectorsyncset_data['metadata']['labels'] = {}
+
+    selectorsyncset_data['metadata']['labels']['managed.openshift.io/osd'] = "true"
+    selectorsyncset_data['metadata']['labels']['managed.openshift.io/gitRepoName'] = arguments.repo_name
```

Comment:
Can/should(?) this and the previous line be moved into the template?

Reply:
Done.

Reply:
These could be moved into the template. The only con would be we would have to check if they exist . Also they only place they get changed is in this code so I am not sure what benefits we would get by moving this there. I had this thought when i wrote this and couldn't decide between which was more proper.